io.c

```diff
@@ -294,13 +294,31 @@ rb_cloexec_open(const char *pathname, int flags, mode_t mode)
     int ret;
     static int o_cloexec_state = -1; /* <0: unknown, 0: ignored, >0: working */
 
+    static const int retry_interval = 0;
+    static const int retry_max_count = 10000;
+
+    int retry_count = 0;
+    int e;
+
 #ifdef O_CLOEXEC
     /* O_CLOEXEC is available since Linux 2.6.23.  Linux 2.6.18 silently ignore it. */
     flags |= O_CLOEXEC;
 #elif defined O_NOINHERIT
     flags |= O_NOINHERIT;
 #endif
-    ret = open(pathname, flags, mode);
+
+    while (1) {
+        ret = open(pathname, flags, mode);
+        e = errno;
+
+        if (ret != -1 || e != EAGAIN || retry_count >= retry_max_count) {
+            break;
+        }
+
+        retry_count++;
+        sleep(retry_interval);
```

Comment:
Is sleeping `retry_interval`(which = 0) intentional?

Reply:
Yes, it is intentional. This will give some CPU time to other threads of the same process and other process.

Reply:
It seems a busy loop, doesn't it consume CPU time until the `open` succeeds?

Reply:
Not so busy since there is a sleep. But sleep 0 to open the file as soon as it is available.